packages/blade/src/components/Dropdown/FilterChipSelectInput.web.tsx

```diff
@@ -113,6 +125,22 @@ const _FilterChipSelectInput = (props: FilterChipSelectInputProps): React.ReactE
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [changeCallbackTriggerer]);
+  useEffect(() => {
+    const isValueEmpty = Array.isArray(value) ? value.length === 0 : !value;
+    if (!isFirstRender && !isValueEmpty) {
```

Comment:
We should avoid doing these things inside useEffect and instead use events. This way you will eliminate the need for checking for isFirstRender hacks